### PR TITLE
docs: clearly mention OpenStack reference implementation

### PR DESCRIPTION
**Pull request by MrBatschner**
**How to categorize this PR?**

/kind documentation
/area os
/os garden-linux

**What this PR does / why we need it**:
This PR adds some documentation which clearly mentions that the current OpenStack feature set is a reference
implementation that is designed for the (not so common) CCEE setup with ESXi as hypervisor.

**Release note**:

```feature user
docs: clearly mention OpenStack reference implementation
```

### CODE REVIEW

**Review by marwinski — CHANGES_REQUESTED** [review]
We agreed to rename the feature in #329. This would probably not fit then!?

**Review by gehoern — APPROVED** [review]

**Comment by MrBatschner** [comment]
@marwinski: we talked about it to put this in for now and @gehoern already approved these changes - once you approve, we can merge it

**Review by marwinski — APPROVED** [review]

**Pull request merged**
